### ISSUE TITLE
publish umbrella artifact with metadata

### ISSUE THREAD

**Issue by yshrsmz** [lightningkite/reacktive#1]
so that what we need become

```
kotlin {
  sourceSets {
    commonMain {
      dependencies {
        // no need to specify dependency for each target
        implementation "com.lightningkite:reacktive:0.1.1"
      }
    }
  }
}
```

looks like current [`reaktive`](https://dl.bintray.com/lightningkite/com.lightningkite.krosslin/com/lightningkite/reacktive/0.1.1/) artifact does not have module metadata and so does not work as umbrella artifact.

**Comment by UnknownJoe796** [comment]
I've been investigating publishing with metadata, and disabled it purposefully.
To quote myself,

> I've noticed that if you publish a library with Gradle Metadata, and then have another library depend on it that also uses Gradle Metadata, you **must** include the first library by its kotlinMetadata artifact as opposed to doing it individually target by target **and** anyone depending on the second library MUST use Gradle Metadata.  Can anyone else confirm this?

As far as I can tell, using metadata is infectious and cuts off options for users.

Most KotlinX repositories *specifically disable this feature*.  For example, [KotlinX Serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/build.gradle#L173) disables it deliberately.

If this stuff is fixed, then such will be supported.  In the mean time, I will be releasing a plugin called `konvenience` soon, which makes it much easier to include each variant of a dependency.

**Comment by UnknownJoe796** [comment]
Here's a sample of the errors you get when trying to publish with metadata and use without:
```Execution failed for task ':compileKotlinAndroidNativeArm32'.
> Could not resolve all files for configuration ':androidNativeArm32CompileKlibraries'.
   > Could not resolve com.lightningkite:kommon-metadata:0.1.3.
     Required by:
         project :
      > Unable to find a matching variant of com.lightningkite:kommon-metadata:0.1.3: Variant 'metadata-api':
          - Found artifactType 'jar' but wasn't required.
          - Found org.gradle.status 'release' but wasn't required.
          - Required org.gradle.usage 'kotlin-api' and found compatible value 'kotlin-api'.
          - Required org.jetbrains.kotlin.native.target 'android_arm32' but no value provided.
          - Required org.jetbrains.kotlin.platform.type 'native' and found incompatible value 'common'.```

**Comment by yshrsmz** [comment]
Thanks for detailed explanation!

> Most KotlinX repositories _specifically disable_ this feature

Actually, this statement is not accurate. 

As [your link](https://github.com/Kotlin/kotlinx.serialization/blob/master/build.gradle#L173)  states, KotlinX libraries `Disable metadata for non-native modules`. 

In other word they disables metadata in jvm/android/js artifacts to prevent them from using metadata, but provide metadata for native artifacts, so that people do not need to specify cpu architecture.(For example [`kotlinx-serialization-runtime-native`](https://kotlin.bintray.com/kotlinx/org/jetbrains/kotlinx/kotlinx-serialization-runtime-native/0.9.1/) is 'umbrella artifact' which then redirect to cpu-architecture-specific artifacts)

I understand your concern, but at least providing metadata for native artifacts make sense.

**Comment by UnknownJoe796** [comment]
Makes sense.  I'm looking into it further, and I think I can get a solution going which works for both cases.
